frontend/src/app/pages/exercises/exercise/shared/exercise-map/utility/ol-map-interactions-manager.ts

```diff
@@ -0,0 +1,173 @@
+import type VectorLayer from 'ol/layer/Vector';
+import type VectorSource from 'ol/source/Vector';
+import { selectCurrentRole } from 'src/app/state/application/selectors/shared.selectors';
+import type { Interaction } from 'ol/interaction';
+import { defaults as defaultInteractions } from 'ol/interaction';
+import type { Subject } from 'rxjs';
+import { combineLatest, takeUntil } from 'rxjs';
+import { selectExerciseStatus } from 'src/app/state/application/selectors/exercise.selectors';
+import type { Feature } from 'ol';
+import { Collection } from 'ol';
+import type OlMap from 'ol/Map';
+import type { AppState } from 'src/app/state/app.state';
+import type { Store } from '@ngrx/store';
+import { selectStateSnapshot } from 'src/app/state/get-state-snapshot';
+import type { ExerciseStatus, Role } from 'digital-fuesim-manv-shared';
+import { TranslateInteraction } from './translate-interaction';
+import type { PopupManager } from './popup-manager';
+import type { FeatureManager } from './feature-manager';
+
+export class OlMapInteractionsManager {
+    private readonly featureLayers: VectorLayer<VectorSource>[];
+    private readonly trainerInteractions: Interaction[];
+    private translateInteraction: TranslateInteraction;
+    private participantInteractions: Interaction[];
+    private interactions: Collection<Interaction>;
+    private lastStatus: ExerciseStatus | undefined;
+    private lastRole: Role | 'timeTravel' | undefined;
+
+    constructor(
+        private readonly mapInteractions: Collection<Interaction>,
+        private readonly store: Store<AppState>,
+        private readonly popupManager: PopupManager,
+        private readonly olMap: OlMap,
+        private readonly layerFeatureManagerDictionary: Map<
```

Comment:
I do think Dictionary is the better Name because this should be used to look up something, Dictionary makes the only look-up part more clear. Map could imply that changing this changes behaviour